Automattic/wp-calypso
# Playwright: migrate Gutenboarding (happy path) spec.

#### Changes proposed in this Pull Request

This PR proposes to migrate the Gutenboarding spec in Selenium, albeit with a few changes.

Key changes:
- Playwright spec focus on the happy path only for now, meaning no meandering back and forth or selecting options that will not be pursued.
- creation of `GutenboardingFlow` object that holds all actions pertaining to the flow.
- addition of helper methods where necessary to implement this flow.

Background details:

Selenium version of the Gutenboarding flow spec meanders occasionally to select options and/or features that will not be applied for the actually created site (which is Free). Such examples include custom domains, changing of language to `es-ES` and others.

The Playwright version of the spec strips out the meandering to focus on the happy path only:
- unique site name is given;
- free `wordpress.com` domain is chosen;
- free design is chosen;
- font pairing is chosen;
- `Plugin` feature is chosen;
- `Free` plan is chosen at the end.

Minor helper methods and tweaks have also been applied to files such as `SidebarComponent` and `DataHelper`.

#### Testing instructions

1. pull the branch
2. `yarn install`
3. `cd test/e2e`
4. `yarn jest specs/specs-playwright/wp-gutenboarding__create.ts`

Ensure that if you run the test spec locally that all steps (including the site deletion flow) completes, otherwise it is best to delete the site manually.

- [x] Playwright mobile passes.
- [x] Playwright desktop passes.


Related to #
